File: SineWaveGenerator.cs
using System;
using System.Globalization;
using System.IO;

public class SineWaveGenerator
{
    public static void Main(string[] args)
    {
        int totalSamples = args.Length > 0 ? int.Parse(args[0], CultureInfo.InvariantCulture) : 240;
        int amplitude = args.Length > 1 ? int.Parse(args[1], CultureInfo.InvariantCulture) : 100;
        int samplesPerCycle = args.Length > 2 ? int.Parse(args[2], CultureInfo.InvariantCulture) : 24;
        int offset = args.Length > 3 ? int.Parse(args[3], CultureInfo.InvariantCulture) : 1000;
        double noisePercent = args.Length > 4 ? double.Parse(args[4], CultureInfo.InvariantCulture) : 0.15;
        long seedValue;
        long? seed = (args.Length > 5 && long.TryParse(args[5], NumberStyles.Integer, CultureInfo.InvariantCulture, out seedValue))
            ? seedValue
            : (long?)null;
        string outputPath = args.Length > 6 ? args[6] : "sine_values_cs.txt";

        // Random in C# accepts an int seed. Mix the long into an int if provided.
        Random rng = seed.HasValue
            ? new Random(unchecked((int)(seed.Value ^ (seed.Value >> 32))))
            : new Random();

        double noiseAmp = noisePercent * amplitude;

        using (var writer = new StreamWriter(outputPath))
        {
            writer.WriteLine("[");
            for (int t = 0; t < totalSamples; t++)
            {
                double value = offset
                               + amplitude * Math.Sin(2.0 * Math.PI * t / samplesPerCycle)
                               + (rng.NextDouble() * 2.0 - 1.0) * noiseAmp; // ±noise
                int rounded = (int)Math.Round(value);
                if (t < totalSamples - 1)
                {
                    writer.WriteLine("  " + rounded + ",");
                }
                else
                {
                    writer.WriteLine("  " + rounded);
                }
            }
            writer.WriteLine("]");
        }

        Console.WriteLine($"Wrote {totalSamples} values to {Path.GetFullPath(outputPath)}");
    }
}

File: SineWaveGenerator.java
import java.util.Random;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.charset.StandardCharsets;

public class SineWaveGenerator {
    public static void main(String[] args) throws Exception {
        int totalSamples = args.length > 0 ? Integer.parseInt(args[0]) : 240;
        int amplitude = args.length > 1 ? Integer.parseInt(args[1]) : 100;
        int samplesPerCycle = args.length > 2 ? Integer.parseInt(args[2]) : 24;
        int offset = args.length > 3 ? Integer.parseInt(args[3]) : 1000;
        double noisePercent = args.length > 4 ? Double.parseDouble(args[4]) : 0.15;
        Long seed = null;
        if (args.length > 5 && !args[5].isEmpty()) {
            try {
                seed = Long.parseLong(args[5]);
            } catch (NumberFormatException ignored) {
                seed = null;
            }
        }
        String outputPath = args.length > 6 ? args[6] : "sine_values.txt";

        Random random = (seed == null) ? new Random() : new Random(seed);
        double noiseAmp = noisePercent * amplitude;

        StringBuilder sb = new StringBuilder();
        sb.append("[\n");
        for (int t = 0; t < totalSamples; t++) {
            double value = offset
                    + amplitude * Math.sin(2.0 * Math.PI * t / samplesPerCycle)
                    + (random.nextDouble() * 2.0 - 1.0) * noiseAmp; // ±noise
            int rounded = (int) Math.round(value);
            sb.append("  ").append(rounded);
            if (t < totalSamples - 1) sb.append(",\n");
        }
        sb.append("\n]\n");

        Path out = Paths.get(outputPath);
        Files.write(out, sb.toString().getBytes(StandardCharsets.UTF_8));
        System.out.println("Wrote " + totalSamples + " values to " + out.toAbsolutePath());
    }
}